qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -545,6 +545,30 @@ def test_snap_schedule_for_number_of_snaps_retention(self):
 
         self.mount_a.run_shell(['rmdir', test_dir[1:]])
 
+    def test_failure_for_duplicate_retention(self):
+        """
+        Test that adding retention for same spec fails for second time.
+        """
+        test_dir = '/' + TestSnapSchedules.TEST_DIRECTORY
+
+        self.mount_a.run_shell(['mkdir', '-p', test_dir[1:]])
+
+        self.fs_snap_schedule_cmd('add', path=test_dir, snap_schedule='1m')
+        self.fs_snap_schedule_cmd('retention', 'add', path=test_dir,
+                                  retention_spec_or_period='m',
+                                  retention_count='50')
+
+        # Adding a duplicate retention spec should fail
+        with self.assertRaises(CommandFailedError):
```

Comment:
here

Reply:
Agree! We can add someting like: `self.assertEqual(exception status, code)` to do that.

Reply:
Sounds like what @mchangir is telling here makes sense. @dparmar18 @neesingh-rh What's pending?